### PR TITLE
Refactor pytorch mnist

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*

*Description of changes:*
1. Split training and inference into different notebooks
2. entry points `train.py` and `inference.py` reside in `code/`
3. use samples from test set to validate the endpoints
4. when testing the endpoint, we download mnist directly from Yann LeCun's website and convert it to numpy array of appropriate shapes depending on the deep learning framework

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by hsl89** [comment]
messed up commits, see #1692
